### PR TITLE
Finished chapter 6 

### PR DESCRIPTION
**Pull request by danimorjan**

### CODE REVIEW

File: online-shop/.gitignore [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# See http://help.github.com/ignore-files/ for more about ignoring files.
+
+# Compiled output
+/dist
```
Comment (teodoraalexandra) [inline online-shop/.gitignore:4]: Suggestion: Add here also .angular and .vscode.

Regarding structure, there is no need to do an additional "wrap" for your application (AngularTraining/online-shop/src). By "wrapper" I'm reffering to the online-shop. It should be enough to habe AngularTraining/src, and also package.json/ package-lock.json and node_modules will be placed at "first" path. There is no mandatory change, just to take into consideration for future (also, will be easier for you too 😄 )

File: online-shop/.vscode/extensions.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+  // For more information, visit: https://go.microsoft.com/fwlink/?linkid=827846
+  "recommendations": ["angular.ng-template"]
```
Comment (teodoraalexandra) [inline online-shop/.vscode/extensions.json:3]: Do you see now the benefit from my prev comment? If they were added in .gitignore, they won't appear at all in changed files.

File: online-shop/src/app/app.component.scss [hunk 3]
```diff
@@ -0,0 +1,15 @@
+.main_container{
+
+    height: 100%;
+
+    width: 65%; 
+
```
Comment (teodoraalexandra) [inline online-shop/src/app/app.component.scss:6]: I suppose you deleted some rows from here that's why it looks like this .. But please remove the empty rows to look more compressed.

File: online-shop/src/app/components/containers/product-details/product-details.component.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+@Component({
+  selector: 'app-product-details',
+  templateUrl: './product-details.component.html',
+  styleUrls: ['./product-details.component.scss']
+})
+export class ProductDetailsComponent {
+  selectedProduct: Product = products[0];
+
+  ngOnInit() {
```
Comment (teodoraalexandra) [inline online-shop/src/app/components/containers/product-details/product-details.component.ts:13]: Please remove the non-used ngOnInit()

File: online-shop/src/app/modules/shopping-cart/components/presentational/shopping-cart-details-view/shopping-cart-details-view.component.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+  ngOnInit() {
+    this.orderProducts = this.products.map((product) => ({
+      product,
+      quantity: 1,
+    }));
+  }
+
+  onRemoveButtonClick(orderProduct: OrderProduct) {
+    const index = this.orderProducts.indexOf(orderProduct);
```
Comment (teodoraalexandra) [inline online-shop/src/app/modules/shopping-cart/components/presentational/shopping-cart-details-view/shopping-cart-details-view.component.ts:21]: Your version is also correct, but I always had a had time in reading the logic of "-1". Do not seems that intuitive to me, but just a personal opinion :) I prefer something like this:

```
if (this.orderProducts.includes(orderProduct)) {
  this.orderProducts.splice(this.orderProducts.indexOf(orderProduct), 1);
}
```

**Comment by teodoraalexandra** [comment]
Also, for small components (where you do not add style and html has only 1 line), you can "compress" them and let only 1 .ts file. But this is a suggestion for future, no need to modify now :) (Same for test files that were generated automatically)

File: online-shop/.vscode/extensions.json (hunk 2 above)
Comment (danimorjan) [reply online-shop/.vscode/extensions.json:3]: Yes. Thank you for the suggestion. I forgot to add them in gitignore.

File: online-shop/src/app/components/containers/product-details/product-details.component.ts (hunk 4 above)
Comment (danimorjan) [reply online-shop/src/app/components/containers/product-details/product-details.component.ts:13]: done

File: online-shop/src/app/app.component.scss (hunk 3 above)
Comment (danimorjan) [reply online-shop/src/app/app.component.scss:6]: i had some deleted rows yes, done

File: online-shop/src/app/modules/shopping-cart/components/presentational/shopping-cart-details-view/shopping-cart-details-view.component.ts (hunk 5 above)
Comment (danimorjan) [reply online-shop/src/app/modules/shopping-cart/components/presentational/shopping-cart-details-view/shopping-cart-details-view.component.ts:21]: I understand, i changed it

File: online-shop/.gitignore (hunk 1 above)
Comment (danimorjan) [reply online-shop/.gitignore:4]: I understand, i made it like this when I created the empty repository. and i noticed now. I will be more careful in the future

**Review by teodoraalexandra — APPROVED** [review]

**Comment by teodoraalexandra** [comment]
Approved, you can merge it :)

**Review by teodoraalexandra — APPROVED** [review]

**Pull request merged**